File: SmartRentCar/Models/RentContract.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRentCar.Models
{
    public class RentContract
    {
        [Key]
        public int RentContractId { get; set; }

        [ForeignKey("ContractStatus")]
        public int ContractStatusId { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        [ForeignKey("Car")]
        public int CarId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        [Range(0, double.MaxValue)]
        public decimal TotalCost { get; set; }

        [Range(0, double.MaxValue)]
        public decimal Deposit { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        public ContractStatus ContractStatus { get; set; }
        public User User { get; set; }
        public Car Car { get; set; }
    }
}

File: SmartRentCar/ClientApp/src/app/models/rentContract.ts
import Decimal from "decimal.js";
import { Car } from "./car";

export interface RentContract {
    rentContractId?: number;
    carId: number;
    contractStatusId: number;
    statusName: string;
    car: Car;
    startDate: Date;
    endDate: Date;
    totalCost: number;
    deposit: number;
    contractAddress?: string;
}

export interface RentContractCreate {
    carId: number;
    startDate: Date;
    endDate: Date;
    totalCost: Decimal;
    deposit: Decimal;
    companyId: number;
}

export enum RentContractStatus {
    PendingPayment = 1,          // Ожидание оплаты
    PendingConfirmation = 2,     // Ожидание подтверждения
    Active = 3,                 // Активная аренда
    PendingEarlyEnd = 4,        // Ожидание досрочного завершения
    PendingCompletion = 5,      // Ожидание завершения
    PendingArbitration = 6,     // Ожидание решения арбитра
    Canceled = 7,               // Отменено
    Completed = 8
}


export interface RentContractUpdate {
    rentContractId: number;
    contractStatusId?: number;
    contractAddress?: string;

}

